Branch-SDK/Branch-SDK/BNCEncodingUtils.m

```diff
@@ -199,7 +199,7 @@ + (NSString *)sanitizedStringFromString:(NSString *)dirtyString {
 
 + (NSData *)encodeDictionaryToJsonData:(NSDictionary *)dictionary {
     NSString *jsonString = [BNCEncodingUtils encodeDictionaryToJsonString:dictionary];
-    return [NSData dataWithBytes:[jsonString UTF8String] length:jsonString.length];
+    return [NSData dataWithBytes:jsonString.UTF8String length:strlen(jsonString.UTF8String)];
```

Comment:
For style preference, could you use the other method you described in #157? Specifically, I prefer not using dot notation for method calls, and here you make the same method call (`UTF8String`) twice.